backend/code/m-booking/src/test/java/com/mbooking/service/UserServiceIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+package com.mbooking.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.mbooking.dto.EditProfileDTO;
+import com.mbooking.dto.UserDTO;
+import com.mbooking.exception.ApiAuthException;
+import com.mbooking.exception.ApiException;
+import com.mbooking.model.Admin;
+import com.mbooking.model.Authority;
+import com.mbooking.model.Customer;
+import com.mbooking.model.User;
+import com.mbooking.repository.AuthorityRepository;
+import com.mbooking.repository.CustomerRepository;
+import com.mbooking.repository.UserRepository;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test_h2")
+public class UserServiceIntegrationTest {
+
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserRepository userRepo;
```

Comment:
If you are performing an integration test, I don't see why you need all of these repositories...